### PR TITLE
Test the implementation of "Move dust collection hook to outside of account mutate"

### PR DESCRIPTION
**Pull request by shamb0**
Testcases to validate #8087 |  Move dust collection hook to outside of account mutate

[#8087](https://github.com/paritytech/substrate/pull/8087)

### CODE REVIEW

**Comment by kianenigma** [comment]
Great! You can rebase this PR and merge it directly to #8087, and we can merge this earlier, while the other one goes for audit and further processes.

File: frame/balances/src/tests_reentrancy.rs [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+			},
+		}.assimilate_storage(&mut t).unwrap();
+
+		let mut ext = sp_io::TestExternalities::new(t);
+		ext.execute_with(|| System::set_block_number(1));
+		ext
+	}
+}
+
+decl_tests!{ Test, ExtBuilder, EXISTENTIAL_DEPOSIT }
```
Comment (thiolliere) [inline frame/balances/src/tests_reentrancy.rs:169]: I don't really see the interest of running the test-suite for this configuration but I'm fine with it

**Review by thiolliere — COMMENTED** [review]
LGTM we can merge once https://github.com/paritytech/substrate/pull/8087 is merged

**Comment by thiolliere** [comment]
actually ideally I would have put the test inside tests.rs and I would have make the configuration in test_composite and test_local optionally sending dust to account 1 (if some static similarly to EXISTENTIAL_DEPOSIT).
But I'm fine with how it is.

**Comment by thiolliere** [comment]
closed in favor of https://github.com/paritytech/substrate/pull/8133
(isn't it ?)

**Pull request closed without merging**